src/Solvers/kaczmarz.jl

```diff
@@ -0,0 +1,323 @@
+            )
+        )
+    end
+
+    if !isdefined(logger, :converged)
+        throw(
+            ArgumentError(
+                "LoggerRecipe $(typeof(logger)) does not contain \
+                the field 'converged' and is not valid for a kaczmarz solver."
+            )
+        )
+    end
+
+    # Assuming that max_it is defined in the logger
+    alpha::Float64 = ingredients.alpha 
+    # We assume the user is using compressors to only decrease dimension
+    sample_size::Int64 = compressor.n_rows
+    cols_a = size(A, 2)
+    # Allocate the information in the buffer using the types of A and b
+    compressed_mat = zeros(eltype(A), sample_size, cols_a)
+    compressed_vec = zeros(eltype(b), sample_size) 
+    # Since sub_solver is applied to compressed matrices use here
+    sub_solver = complete_sub_solver(ingredients.sub_solver, compressed_mat, compressed_vec)
+    mat_view = view(compressed_mat, 1:sample_size, :)
+    vec_view = view(compressed_vec, 1:sample_size)
+    solution_vec = x
+    update_vec = zeros(eltype(x), cols_a)
+    return KaczmarzRecipe{
+        eltype(A), 
+        typeof(b), 
+        typeof(A), 
+        typeof(vec_view),
+        typeof(mat_view),
+        typeof(compressor),
+        typeof(logger),
+        typeof(error),
+        typeof(sub_solver)
+    }(
+        compressor, 
+        logger, 
+        error,
+        sub_solver,
+        alpha,
+        compressed_mat,
+        compressed_vec,
+        solution_vec,
+        update_vec,
+        mat_view,
+        vec_view
+    )
+end
+
+"""
+    kaczmarz_update!(solver::KaczmarzRecipe)
+
+A function that performs the Kaczmarz update when the compression dimension is one. 
+    If ``a`` is the resulting compression of the coefficient matrix, 
+    and ``c`` is the resulting compression of the constant vector, 
+    then we perform the update: ``x = x - \\alpha (a^\\top x -c) / \\|a\\|_2^2``. 
+
+# Arguments
+- `solver::KaczmarzRecipe`, the solver information required for performing the update.
+
+# Outputs
+- returns `nothing`
+"""
+function kaczmarz_update!(solver::KaczmarzRecipe)
+    # when the constant vector is a zero dimensional subArray we know that we should perform
+    # the one dimension kaczmarz update
+
+    # Compute the projection scaling (bi - dot(ai,x)) / ||ai||^2
+    scaling = solver.alpha * (dotu(solver.mat_view, solver.solution_vec) 
+        - solver.vec_view[1]) 
+    scaling /= dot(solver.mat_view, solver.mat_view)
+    # udpate the solution
+    axpby!(-scaling, solver.mat_view', 1.0, solver.solution_vec)
+    return nothing
+end
+
+
+"""
+    kaczmarz_update_block!(solver::KaczmarzRecipe)
+
+A function that performs the kaczmarz update when the compression dim is greater than 1.  
+    In the block case where the compressed matrix s ``B``, and the compressed contant 
+    vector ``g``, we perform the updated: 
+    ``x = x - \\alpha B^\\top (BB^\\top)^\\dagger(Bx - g)``.
```

Comment:
Consider renaming `B` and `g` to keep it consistent with the Kacmarz docstring above. Or change the docstring above.